wordpress-mobile/WordPress-iOS
# Spotlight: User activity indexing

This PR adds user activity tracking/indexing for portions of the app:

![activity_indexing](https://user-images.githubusercontent.com/154014/39056564-b8dd7ecc-447c-11e8-8895-41f867e250ef.gif)

Fixes #9148

## Discussion

Similar to what was done in core spotlight item indexing from #8906 and #8945, a new protocol is introduced here (`SearchableActivityConvertable.swift`) that VC's can adopt if they want to index themselves in the on-device spotlight index. Also like before `SearchManager.swift` still handles opening the appropriate VC screen based on the incoming `NSUserActivity`. 

This PR is constrained to indexing the following screens (we can always add more in a subsequent PR):

* The 4 tabs across the bottom (My Sites, Reader, Me, Notifications
* Individual site (main menu)
* App Settings
* Notifications Settings
* Help & Support

### On Indexing NSUserActivity

Screens are indexed as the user sees them. If a user never accesses a screen, they will not be indexed.

Also, NSUserActivity indexing is a little different that Core Spotlight — the big difference is that even if you clear the index in app settings, you will still see the indexed screens in spotlight. A NSUserActivity is only removed if:

* You uninstall the app
* The expiration date passes
* Apple/iOS wants to 😀 

Because of this 👆  the default expiration is set to accessing the screen + one week (and is refreshed every time they access the screen). Hopefully this is a good balance that will keep the most recent screens in spotlight searches.

## Testing

* Test navigating to the screens listed above 👆 and then searching for them in spotlight
* Check out the keywords for each screen (in code) and search for that
* Try using an iPad too

**Note:** The sim seems to work for this, but also test on device as well.

@etoledom would you mind taking a look at this? Thank you!!!
